--- a/mpp/soc.cmake
+++ b/mpp/soc.cmake
@@ -193,6 +193,11 @@ add_soc_config("RV1126B"
     "H264D,H265D,JPEGD"
     "H264E,H265E,JPEGE")
 
+add_soc_config("RV1103, RV1106"
+    "VEPU540C"
+    ""
+    "H264E,H265E,JPEGE")
+
 add_soc_config("RK353[89]"
     "VDPU384B,VDPU730,VDPU2"
     "H263D,H264D,H265D,JPEGD,MPEG2D,MPEG4D,VP8D,VP9D,AV1D,AVSD"
--- a/osal/inc/mpp_soc.h
+++ b/osal/inc/mpp_soc.h
@@ -47,6 +47,8 @@ typedef enum RockchipSocType_e {
     ROCKCHIP_SOC_RK3562,
     ROCKCHIP_SOC_RK3576,
     ROCKCHIP_SOC_RV1126B,
+    ROCKCHIP_SOC_RV1103,
+    ROCKCHIP_SOC_RV1106,
     ROCKCHIP_SOC_RK3538,
     ROCKCHIP_SOC_RK3572,
     ROCKCHIP_SOC_RK3539,
--- a/osal/mpp_soc.c
+++ b/osal/mpp_soc.c
@@ -1060,6 +1060,26 @@ static const MppSocInfo mpp_soc_infos[]
         {   &vepu511, NULL, NULL, NULL},
     },
     {   /*
+         * rv1103 has codec:
+         * 1 - RK H.264/H.265/MJPEG encoder
+         */
+        "rv1103",
+        ROCKCHIP_SOC_RV1103,
+        HAVE_RKVENC,
+        {   NULL, NULL, NULL, NULL, NULL, NULL},
+        {   &vepu540c, NULL, NULL, NULL},
+    },
+    {   /*
+         * rv1106 has codec:
+         * 1 - RK H.264/H.265/MJPEG encoder
+         */
+        "rv1106",
+        ROCKCHIP_SOC_RV1106,
+        HAVE_RKVENC,
+        {   NULL, NULL, NULL, NULL, NULL, NULL},
+        {   &vepu540c, NULL, NULL, NULL},
+    },
+    {   /*
          * RK3538 has codec:
          * 1 - RK H.264/H.265/VP9/AV1 1080P decoder
          * 2 - RK jpeg decoder
--- a/mpp/hal/rkenc/h264e/hal_h264e_vepu540c.c
+++ b/mpp/hal/rkenc/h264e/hal_h264e_vepu540c.c
@@ -1748,6 +1748,8 @@ const MppEncHalApi hal_h264e_vepu540c =
     .soc_type   = {
         ROCKCHIP_SOC_RK3528,
         ROCKCHIP_SOC_RK3562,
+        ROCKCHIP_SOC_RV1103,
+        ROCKCHIP_SOC_RV1106,
         ROCKCHIP_SOC_BUTT
     },
 };
--- a/mpp/hal/rkenc/h265e/hal_h265e_vepu540c.c
+++ b/mpp/hal/rkenc/h265e/hal_h265e_vepu540c.c
@@ -1710,6 +1710,8 @@ const MppEncHalApi hal_h265e_vepu540c =
     .client     = VPU_CLIENT_RKVENC,
     .soc_type   = {
         ROCKCHIP_SOC_RK3528,
+        ROCKCHIP_SOC_RV1103,
+        ROCKCHIP_SOC_RV1106,
         ROCKCHIP_SOC_BUTT
     },
 };
